tests/api/archive_basic_test.rs

```diff
@@ -189,3 +194,300 @@ async fn get_law_pdf_request_with_incorrect_path_expect_not_found() {
+    branch_name: &str,
+    app: &impl Service<Request, Response = ServiceResponse<impl MessageBody>, Error = Error>,
+    expected: bool,
+) {
+    for request_uri in file_paths {
+        let req = test::TestRequest::get()
+            .uri(&format!(
+                "/_stelae/test_org/{}?commitish={}&remainder={}",
+                repo_name, branch_name, request_uri
+            ))
+            .to_request();
+        let resp = test::call_service(&app, req).await;
+        let actual = if expected {
+            resp.status().is_success()
+        } else {
+            resp.status().is_client_error()
+        };
+        let expected = true;
+        assert_eq!(actual, expected);
+    }
+}
+
+#[actix_web::test]
+async fn test_resolve_root_stele_law_html_different_files_with_different_branches() {
+    let archive_path =
+        common::initialize_archive_without_bare(ArchiveType::Basic(Jurisdiction::Single)).unwrap();
+    let app = common::initialize_app(archive_path.path()).await;
+
+    let mut path = archive_path.path().to_path_buf();
+    path.push("test_org");
+    let git_repo = get_repository(&path, "law-html");
+    path.push("law-html");
+    let _ = git_repo.create_branch("test_branch");
+
+    let _ = git_repo.checkout("master");
+    let _ = git_repo.add_file(&path, "test.txt", "Content for master branch");
+    let _ = git_repo.commit(None, "Adding data for master branch");
+
+    let _ = git_repo.checkout("test_branch");
+    let _ = git_repo.add_file(&path, "test1.txt", "Content for test branch");
+    let _ = git_repo.commit(None, "Adding data for master branch");
+
+    let req = test::TestRequest::get()
+        .uri(&format!(
+            "/_stelae/test_org/law-html?commitish=master&remainder=/test.txt"
+        ))
+        .to_request();
+    let resp = test::call_service(&app, req).await;
+    let actual = resp.status().is_success();
+    assert_eq!(actual, true);
+
+    let req = test::TestRequest::get()
+        .uri(&format!(
+            "/_stelae/test_org/law-html?commitish=master&remainder=/test1.txt"
+        ))
+        .to_request();
+    let resp = test::call_service(&app, req).await;
+    let actual = resp.status().is_success();
+    assert_eq!(actual, false);
+
+    let req = test::TestRequest::get()
+        .uri(&format!(
+            "/_stelae/test_org/law-html?commitish=test_branch&remainder=/test.txt"
+        ))
+        .to_request();
+    let resp = test::call_service(&app, req).await;
+    let actual = resp.status().is_success();
+    assert_eq!(actual, false);
+
+    let req = test::TestRequest::get()
+        .uri(&format!(
+            "/_stelae/test_org/law-html?commitish=test_branch&remainder=/test1.txt"
+        ))
+        .to_request();
+    let resp = test::call_service(&app, req).await;
+    let actual = resp.status().is_success();
+    assert_eq!(actual, true);
+}
+
+#[actix_web::test]
+async fn test_resolve_root_stele_law_html_file_content_with_different_branches_expect_success() {
```

Comment:
suggestion: we'll also need a couple of tests when an archive is initialized as multihost, see [test_archive_multihost.rs](https://github.com/openlawlibrary/stelae/blob/main/tests/api/archive_multihost_test.rs). so calling out to `_stelae` endpont for `stele_1` and `stele_2` and expecting content to be resolved

suggestion 2: a test testing branch with a `/` in it would also be useful, I think.